### ISSUE TITLE
Missing files

### ISSUE THREAD

**Issue by campoy** [src-d/datasets#26]
While developing `pga` I noticed that many files in the index appear missing from `/siva/latest`

For instance:

```bash
could not get e9477061b9e1b80c2ac5c96abcb8d236c11549a1.siva: could not get http://pga.sourced.tech/siva/latest/e9/e9477061b9e1b80c2ac5c96abcb8d236c11549a1.siva: 404 Not Found
could not get ebd6182d6d724d79fba1bf0cf28c7ae6706382f3.siva: could not get http://pga.sourced.tech/siva/latest/eb/ebd6182d6d724d79fba1bf0cf28c7ae6706382f3.siva: 404 Not Found
could not get e76a3aaa800670389d19bd2b84b9489455d883d8.siva: could not get http://pga.sourced.tech/siva/latest/e7/e76a3aaa800670389d19bd2b84b9489455d883d8.siva: 404 Not Found
could not get e7a4ab4093dcf8423a14fd20ee58f073e90f2ae4.siva: could not get http://pga.sourced.tech/siva/latest/e7/e7a4ab4093dcf8423a14fd20ee58f073e90f2ae4.siva: 404 Not Found
```

The way I created these URLs was by using the prefix `http://pga.sourced.tech/siva/latest/` plus the first to characters in the siva filename as a directory, plus the siva filename at the end.

Is that wrong or do we have another issue?

@vmarkovtsev

**Comment by vmarkovtsev** [comment]
Redirect this to @ajnavarro

**Comment by ajnavarro** [comment]
The csv is newer than the siva syncronization. We are going to do the last sync this week with a new CSV and a new set of siva files.

**Comment by ajnavarro** [comment]
Right now, we are generating a new CSV version. It takes hours, so tomorrow we will sync all the new files to the PGA server.

**Comment by campoy** [comment]
could you write a little post-mortem explaining exactly what happened and the measures we'll take to avoid this in the future?
thanks!

**Comment by ajnavarro** [comment]
Basically, generate and upload a 50 mb CSV file take less than upload 3Tb of siva files.

To upload the siva files, we generated at some point of the time a list of paths to upload to the server. Meanwhile borges was processing more repositories and creating new siva files. This took like 2-3 days to upload to the server. At the same time, we generated 3 CSV versions, one more updated that the previous one. Generate and upload that file takes only hours.

That's why you have files on the CSV that are not uploaded to the PGA server.

Ways from the top of my head to avoid this problem on the future are:
- Wait until have all the files downloaded to create the CSV file and upload the siva files at the same time than the CSV.
- Create a more complicated script to only generate the csv from a specific date and time. Do the same with the siva upload process.
